appeals/web/src/server/appeals/appeal-details/application-submission-date/application-submission-date.controller.js

```diff
@@ -0,0 +1,118 @@
+import logger from '#lib/logger.js';
+import { objectContainsAllKeys } from '#lib/object-utilities.js';
+import { addNotificationBannerToSession } from '#lib/session-utilities.js';
+import { getAppellantCaseFromAppealId } from '../appellant-case/appellant-case.service.js';
+import { changeApplicationSubmissionDatePage } from './application-submission-date.mapper.js';
+import { changeApplicationSubmissionDate } from './application-submission-date.service.js';
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getChangeApplicationSubmissionDate = async (request, response) => {
+	return renderChangeApplicationSubmissionDate(request, response);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderChangeApplicationSubmissionDate = async (request, response) => {
+	try {
+		const { errors, currentAppeal, apiClient } = request;
+		const appellantCaseData = await getAppellantCaseFromAppealId(
+			apiClient,
+			currentAppeal.appealId,
+			currentAppeal.appellantCaseId
+		).catch((error) => {
+			logger.error(error);
+			return response.status(404).render('app/404.njk');
+		});
```

Comment:
Ideally we don't mix `try/catch` with `.then/.catch` - here we are returning a 404 response, but the rest of the function will continue to execute.